widget/widget_renderer.go

```diff
@@ -0,0 +1,34 @@
+package widget
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// When using the baseRenderer the embedding renderer should call
+// layoutShadow(contentSize, contentPos) to lay out the shadow.
+type baseRenderer struct {
+	objects []fyne.CanvasObject
+	sh      fyne.CanvasObject
```

Comment:
Can we full name this to shadow?

Reply:
Of course! This is an artifact from preparations I made.